ecommerce-ui/src/App.js

```diff
@@ -0,0 +1,56 @@
+import React, {Component } from 'react';
+import CardHolder from './components/cardHolder/cardHolder';
+import Cart from './components/cart/cart';
+import PropTypes from 'prop-types';
+import './App.css';
+
+class App extends Component {
+  state = {
+    sites: [],
+    reservedSites: [],
+    resCount: 0,
+  }
+
+  toggleReserve = (idx, title) => {
+    return () => {
+      const reservation = title;
+//      const reservation = this.state.sites[idx];
+
+      if(this.state.reservedSites.includes(reservation)) {
+        alert("You already booked that one.");
+        return;
+      }
+
+      this.setState(prevState => {
+        console.log(this.state.reservedSites.length);
+        return {
+          reservedSites: [...prevState.reservedSites, reservation],
+          resCount: this.state.reservedSites.length+1 //why do I need to add one?
```

Comment:
I assume you wanted to update the amount of reserved sites by passing in the length of the array. Remember that at this point state has not yet updated. this return must resolve first. I would also recommend using prevState.reservedSites.length +1 
prev state is static while this.state could theoretically be a different value.